src/main/kotlin/uk/gov/justice/digital/hmpps/pollpush/config/RestTemplateConfiguration.kt

```diff
@@ -53,7 +56,10 @@ class RestTemplateConfiguration(private val apiDetails: ClientCredentialsResourc
 
 class NotFoundAndConflictIgnoringResponseErrorHandler : DefaultResponseErrorHandler() {
   // ignore not found and conflict errors as they are very frequent and fill up the logs
-  // don't need to retry on those errors either
-  override fun hasError(statusCode: HttpStatus): Boolean =
-      statusCode.isError && statusCode != HttpStatus.NOT_FOUND && statusCode != HttpStatus.CONFLICT
+  // don't need to retry on those errors either  override fun handleError(response: ClientHttpResponse) {
+  override fun handleError(response: ClientHttpResponse) {
+    if (response.statusCode == HttpStatus.NOT_FOUND || response.statusCode == HttpStatus.CONFLICT) return
```

Comment:
turns out that the `OAuth2ErrorHandler` overrides the `hasError` and doesn't delegate if it is a client error, just handles it instead